tests/st/libnetwork/test_mainline_multi_host.py

```diff
@@ -46,46 +51,95 @@ def test_multi_host(self):
-            workload_host2 = host2.create_workload("workload2",
-                                                   network=network)
+            # Create the networks on host1, but it should be usable from all
+            # hosts.  We create one network using the default driver, and the
+            # other using the Calico driver.
+            network1 = host1.create_network("testnet1", ipam_driver="default")
+            network2 = host1.create_network("testnet2", ipam_driver="calico")
+
+            # Assert that the networks can be seen on host2
+            assert_network(host2, network2)
+            assert_network(host2, network1)
+
+            # Assert that the profiles have been created for the networks
+            profile_name1 = get_profile_name(host1, network1)
+            assert_profile(host1, profile_name1)
+            profile_name2 = get_profile_name(host1, network2)
+            assert_profile(host1, profile_name2)
+
+            # Create two workloads on host1 and one on host2 all in network 1.
+            workload_h1n2_1 = host1.create_workload("workload_h1n2_1",
+                                                    network=network2)
+            workload_h1n2_2 = host1.create_workload("workload_h1n2_2",
+                                                    network=network2)
+            workload_h2n2_1 = host2.create_workload("workload_h2n2_1",
+                                                    network=network2)
+
+            # Create similar workloads in network 2.
+            workload_h2n1_1 = host2.create_workload("workload_h2n1_1",
+                                                    network=network1)
+            workload_h1n1_1 = host1.create_workload("workload_h1n1_1",
+                                                    network=network1)
+            workload_h1n1_2 = host1.create_workload("workload_h1n1_2",
+                                                    network=network1)
 
             # Assert that endpoints are in Calico
-            assert_number_endpoints(host1, 1)
-            assert_number_endpoints(host2, 1)
-
-            # Assert that workloads can communicate with each other
-            workload_host1.assert_can_ping(workload_host2.ip, retries=5)
-            self.assert_connectivity(pass_list=[workload_host1,
-                                                workload_host2])
-            # Ping using container names
-            workload_host1.execute("ping -c 1 -W 1 workload2")
-            workload_host2.execute("ping -c 1 -W 1 workload1")
+            assert_number_endpoints(host1, 4)
+            assert_number_endpoints(host2, 2)
+
+            # Assert that workloads can communicate with each other on network
+            # 1, and not those on network 2.  Ping using IP for all workloads,
+            # and by hostname for workloads on the same network (note that
+            # a workloads own hostname does not work).
+            workload_h1n1_1.execute("cat /etc/hosts")
```

Comment:
I hadn't intended to take them out - thought it provides some useful diags incase host name stuff ever goes wrong.  But I can take it out if you'd prefer.